pdns/tcpiohandler.hh

```diff
@@ -208,9 +208,8 @@ public:
       return d_conn->tryRead(buffer, pos, toRead);
     }
 
```

Comment:
The cases `toRead == pos` (will cause `::read` to rerturn `0`) and `pos > toRead` (sign wrap) should be handled.